### DISCUSSION TITLE
Custom webpack config

### DISCUSSION THREAD

**Discussion by olegchursin** [studds/nx-aws#61] [Q&A] [answered]
Is there a way to pass custom webpack config?

**Comment by studds** [comment, marked as answer]
Hey @olegchursin, yes there is, the `webpackConfig` option (which is the same as the option available on [`@nrwl/node:build`](https://nx.dev/latest/angular/plugins/node/executors/build#webpackconfig):

`angular.json` / `workspace.json`:

```json
"build": {
      "builder": "@nx-aws/sam:build",
      "options": {
          "outputPath": "dist/apps/api",
          "template": "apps/api/template.yaml",
          "tsConfig": "apps/api/tsconfig.app.json",
          "assets": ["apps/api/src/assets"],
          "webpackConfig": "apps/api/src/webpack.config.js",
          "sourceMap": true,
          "maxWorkers": 14
      }
  },
```

And then `apps/api/src/webpack.config.js` is a function which takes a webpack config, and returns the enhanced / replaced webpack config:

```js
const { merge } = require('webpack-merge');

module.exports = (initialConfig) => {
    return merge(initialConfig, {
        // your options here
    });
};

```

**Reply by olegchursin** [reply]
Thanks for a quick response and for your work on this package @studds !
